### PR TITLE
Dashboard: GUI reflects DISABLE_SEARCH_INDEXING

### PR DESCRIPTION
**Pull request by jrwdunham**
If Archivematica has been deployed with Elasticsearch indexing disabled, then the following changes will result:

1. The Backlog, Appraisal, and Archival storage tabs will now no longer be displayed in the navigation bar. If the user navigates to them using their URLs, the pages will indicate their non-operationality.
2. The SIP Arrange pane of the Ingest tab will no longer be displayed.
3. The disabled status of ES indexing will be indicated under Administration > General.
4. The "Send to backlog" option of "Create SIP(s)" will not be available in the Processing Configuration edit pages.
5. When the "Create SIP(s)" choice is displayed to the user during transfer, it will no longer contain a "Send to backlog" option.

Fixes https://github.com/artefactual/archivematica/issues/803 and implements feature file at https://github.com/artefactual-labs/archivematica-acceptance-tests/pull/50.

Possible TODO: If the user supplies a processingConfig.xml file in a transfer that causes "Create SIP(s)" to default to "Send to backlog", then MCPServer should silently ignore this.

### CODE REVIEW

**Comment by sevein** [comment]
@jrwdunham we can't merge this into qa/1.x because AM17 hasn't been released yet. Should't you be targetting dev/1.8.x instead?

File: src/MCPServer/lib/linkTaskManagerChoice.py [hunk 1]
```diff
@@ -57,6 +57,9 @@ def __init__(self, jobChainLink, pk, unit):
 
         choices = MicroServiceChainChoice.objects.filter(choiceavailableatlink_id=str(jobChainLink.pk))
         for choice in choices:
+            if (django_settings.DISABLE_SEARCH_INDEXING and
+                    choice.chainavailable.description == 'Send to backlog'):
```
Comment (sevein) [inline src/MCPServer/lib/linkTaskManagerChoice.py]: Could we instead have a number of abilities and hide these items based on that? E.g. pipeline `X` has the `search` ability disabled, choice `Y` depends on `search` ability so it's ignored.

File: src/dashboard/src/templates/appraisal/appraisal.html [hunk 2]
```diff
@@ -20,9 +20,14 @@
...
-    <ng-include src="'front_page/content.html'"></ng-include>
-</div>
-{% endverbatim %}
+  {% if disable_search_indexing %}
+    <h1>Elasticsearch Indexing Disabled</h1>
+    <p class="es-indexing-disabled">Sorry, Elasticsearch indexing has been disabled in this Archivematica installation. The appraisal tab is non-functional when indexing is turned off.</p>
+  {% else %}
+    {% verbatim %}
+    <div ng-app="appraisalTab">
+          <ng-include src="'front_page/content.html'"></ng-include>
```
Comment (sevein) [inline src/dashboard/src/templates/appraisal/appraisal.html]: Fix indentation.

File: src/dashboard/src/templates/archival_storage/list.html [hunk 3]
```diff
@@ -25,103 +25,110 @@
...
 
-  <ul class="breadcrumb">
-    {% trans "Archival storage" as archival_storage_label %}
-    {% trans "Search" as search_label %}
-    {% breadcrumb_url archival_storage_label 'components.archival_storage.views.overview' %}
-    {% breadcrumb search_label %}
-  </ul>
+  {% if disable_search_indexing %}
+    <h1>Elasticsearch Indexing Disabled</h1>
+    <p class="es-indexing-disabled">Sorry, Elasticsearch indexing has been disabled in this Archivematica installation. The archival storage tab is non-functional when indexing is turned off.</p>
```
Comment (sevein) [inline src/dashboard/src/templates/archival_storage/list.html]: Fix i18n.

**Review by sevein — CHANGES_REQUESTED** [review]

File: src/MCPServer/lib/linkTaskManagerChoice.py (hunk 1 above)
Comment (jrwdunham) [reply src/MCPServer/lib/linkTaskManagerChoice.py]: Yes, I think that's a good idea. I've tried to implement what you've described in the most recent commit. Let me know what you think.

File: src/dashboard/src/templates/appraisal/appraisal.html (hunk 2 above)
Comment (jrwdunham) [reply src/dashboard/src/templates/appraisal/appraisal.html]: fixed

File: src/dashboard/src/templates/archival_storage/list.html (hunk 3 above)
Comment (jrwdunham) [reply src/dashboard/src/templates/archival_storage/list.html]: Fixed here and also in all newly added template strings.

**Comment by sevein** [comment]
@jrwdunham, I've rebased `dev/1.8.x` because it was behind qa/1.x missing some important fixes.

I've also created `stable/1.7.x` so you can start submitting pull requests against `qa/1.x` for AM18 stuff. Perhaps you can repurpose this PR to include what's in `dev/1.8.x` plus changes in `dev/issue-11442-indexless-GUI`. We should delete `dev/1.8.x`, `dev/1.8.x-indexless-GUI` and `dev/1.8.x-without-logs` I assume?

**Comment by jrwdunham** [comment]
@sevein I've rebased this against the new dev/1.8.x. Can you CR again?

The dev/1.8.x-indexless-GUI and dev/1.8.x-without-logs branches are gone.

**Review by sevein — CHANGES_REQUESTED** [review]
Two observations:

- I think that `Add common abilities with choice dependencies` should be updated
so it doesn't include the setting rename because it's unrelated.

- Regarding to `Dashboard: GUI reflects DISABLE_SEARCH_INDEXING`, I think as a
developer my only setting interface should be `django.conf.settings`. I don't
want to remember what `get_search_enabled` is, I want to simply look up
`django_settings.SEARCH_ENABLED`. I think your motivation is good because the
existing name sucks but I'm not confortable solving that problem this way
because it's not cohesive. IMO it would be a win win (for users and for us) to
implement it in `appconfig.Config` and add some tests - and if you don't have
time for that then I would just ignore this for now. Can we have something like
this in `CONFIG_MAPPING`?

```
'search_enabled': [
    {'section': 'MCPClient', 'option': 'disableElasticsearchIndexing', 'type': 'boolean'},
    {'section': 'MCPClient', 'option': 'search_enabled', 'type': 'iboolean'},
}
```

Two changes would be required:

- `appconfig.Config.get` would need to handle lists.
- `env_configparser.EnvConfigParser` would need to have a new `iboolean` method
added (you could subclass `EnvConfigParser` too).

File: src/MCPClient/lib/settings/common.py [hunk 4]
```diff
@@ -58,6 +61,8 @@
     'db_port': {'section': 'client', 'option': 'port', 'type': 'string'},
 }
 
+SEARCH_ENABLED_DEFAULT = False
```
Comment (sevein) [inline src/MCPClient/lib/settings/common.py]: Do we still need this?

File: src/archivematicaCommon/lib/env_configparser.py [hunk 5]
```diff
@@ -70,3 +70,7 @@ def getfloat(self, *args, **kwargs):
     @fallback_option
     def getboolean(self, *args, **kwargs):
         return ConfigParser.SafeConfigParser.getboolean(self, *args, **kwargs)
+
+    @fallback_option
+    def getiboolean(self, *args, **kwargs):
+        return not self.getboolean(*args, **kwargs)
```
Comment (sevein) [inline src/archivematicaCommon/lib/env_configparser.py:76]: `tests/test_env_configparser.py` has already good coverage, should we include a test for this new method?

Comment (jrwdunham) [reply src/archivematicaCommon/lib/env_configparser.py:76]: I've added new tests for appconfig that test the new "multiple equivalent options" feature.

File: src/MCPClient/lib/settings/common.py (hunk 4 above)
Comment (jrwdunham) [reply src/MCPClient/lib/settings/common.py]: No, I guess not

**Review by sevein — APPROVED** [review]

**Pull request merged**